packages/block-editor/src/components/colors/use-colors.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * External dependencies
+ */
+import memoize from 'memize';
+import { startCase } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Children, cloneElement, useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import InspectorControls from '../inspector-controls';
+import PanelColorSettings from '../panel-color-settings';
+import { useBlockEditContext } from '../block-edit';
+
+const createComponent = memoize( ( attribute, color ) => ( { children } ) =>
+	// Clone children, setting the style attribute from the color configuration,
+	// if not already set explicitly through props.
+	Children.map( children, ( child ) =>
+		cloneElement( child, {
+			style: { [ attribute ]: color, ...child.props.style },
+		} )
+	)
+);
+
+const createSetColor = memoize( ( setAttributes, name ) => ( newColor ) =>
```

Comment:
I guess we may have a memory leak here, each time setAttributes changes, we memoize another object, never cleaning old references.
Do we really need this memoize? The setters are already being created inside a useMemo hook.

Reply:
Yes, otherwise a single attribute change would recreate all setters. `setAttributes` shouldn't change often and we can always set a `maxSize`.